models/reg_lstm/__main__.py

```diff
@@ -79,20 +80,28 @@ def evaluate_dataset(split_name, dataset_cls, model, embedding, loader, batch_si
         'Reuters': Reuters,
         'AAPD': AAPD,
         'IMDB': IMDB,
-        'Yelp2014': Yelp2014
+        'Yelp2014': Yelp2014,
+        'AG_NEWS': AGNews
     }
 
     if args.dataset not in dataset_map:
         raise ValueError('Unrecognized dataset')
 
     else:
         dataset_class = dataset_map[args.dataset]
-        train_iter, dev_iter, test_iter = dataset_class.iters(args.data_dir,
-                                                              args.word_vectors_file,
-                                                              args.word_vectors_dir,
-                                                              batch_size=args.batch_size,
-                                                              device=args.gpu,
-                                                              unk_init=UnknownWordVecCache.unk)
+        iters = dataset_class.iters(args.data_dir,
+                                    args.word_vectors_file,
+                                    args.word_vectors_dir,
+                                    batch_size=args.batch_size,
+                                    device=args.gpu,
+                                    unk_init=UnknownWordVecCache.unk)
+
+        # Some datasets (e.g. AG_NEWS) only have train and test splits
+        if len(iters) == 2:
+            train_iter, test_iter = iters
+            dev_iter = test_iter
```

Comment:
if the dataset only has 2 splits (train, test), then we set dev_iter = test_iter, so that it can print evaluation metrics while training